components/support/sql/src/open_database.rs

```diff
@@ -22,6 +22,8 @@
 ///        it and call prepare(), upgrade_from() for each upgrade that needs to be applied, then
 ///        finish(). As above, a read-only connection will panic if upgrades are necessary, so
 ///        you should ensure the first connection opened is writable.
+///      - If the database file is corrupt, or your upgrade returns [`Error::Corrupted`], the
+///        database file will be removed and replaced with a new DB.
```

Comment:
note: Desktop does this in a few...places 😅 already, and I thought it's a nice pattern to expose more generally!

* [Places](https://searchfox.org/mozilla-central/rev/dcf50463d3e1e1f3a91bc1406e3772b2a5f62774/toolkit/components/places/Database.cpp#1112-1116) will replace databases from Firefox versions [older than the last ESR before the most recent watershed release](https://bugzilla.mozilla.org/show_bug.cgi?id=1847592).
* The [bookmarks mirror](https://searchfox.org/mozilla-central/rev/dcf50463d3e1e1f3a91bc1406e3772b2a5f62774/toolkit/components/places/SyncedBookmarksMirror.sys.mjs#1536-1541) throws away any databases from when the mirror was pref'd off, instead of trying to migrate them.

Reply:
```suggestion
///      - If the database file is corrupt, or your upgrade returns [`Error::Corrupt`], the
///        database file will be removed and replaced with a new DB.
```

Reply:
Agree "Corrupt" isn't great - I really don't see a reason you can't reuse `IncompatibleVersion` here though?

Reply:
Oooh, good question! I wasn't sure about throwing the database away in case of an incompatible version—that felt like an error that we'd want to know about, especially since there's no way to downgrade a profile on mobile, and other migration errors don't trigger automatic replacement. But I'd also be okay with having `IncompatibleVersion` trigger the replacement behavior!